src/Analyzers/CSharp/CodeFixes/UsePatternMatching/CSharpUseNotPatternCodeFixProvider.cs

```diff
@@ -65,14 +65,27 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
 
             var notExpression = (PrefixUnaryExpressionSyntax)notExpressionLocation.FindNode(getInnermostNodeForTie: true, cancellationToken);
             var parenthesizedExpression = (ParenthesizedExpressionSyntax)notExpression.Operand;
-            var isPattern = (IsPatternExpressionSyntax)parenthesizedExpression.Expression;
+            var oldExpression = parenthesizedExpression.Expression;
+
+            var updatedPattern = oldExpression switch
+            {
+                IsPatternExpressionSyntax isPattern => isPattern.WithPattern(UnaryPattern(Token(SyntaxKind.NotKeyword), isPattern.Pattern)),
+                BinaryExpressionSyntax
+                {
+                    Right: TypeSyntax type
+                } binaryIsExpression => IsPatternExpression(binaryIsExpression.Left, UnaryPattern(Token(SyntaxKind.NotKeyword), TypePattern(type))),
+                _ => null
```

Comment:
how does this case get hit?

Reply:
It's required for successful CI build.

Reply:
@reacheight You can use `throw ExceptionUtilities.Unreachable;` instead.